### ISSUE TITLE
Client sends too many Ctrl's

### ISSUE THREAD

**Issue by jbarber** [rainforestapp/Flashlight-vnc#22]
Using the following program:
http://delphiforfun.org/programs/utilities/KeyCodes.htm

we can see that a single press of the 'a' key results in the following key events:
```
OnKeyDown, Key code=255, Control keys=, Key name
OnKeyup, Key code=17, Control keys=, Key name CTRL
OnKeyup, Key code=17, Control keys=, Key name CTRL
OnKeyDown, Key code=65, Control keys=, Key name a
OnkeyPress a
OnKeyup, Key code=17, Control keys=, Key name CTRL
OnKeyup, Key code=64, Control keys=, Key name CTRL
OnKeyup, Key code=255, Control keys=, Key name
```

'A' results in:
```
OnKeyDown, Key code=16, Control keys=Shift{Left}, Key name SHIFT
OnKeyDown, Key code=255, Control keys=Shift{Left}, Key name
OnKeyup, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyup, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyDown, Key code=65, Control keys=Shift{Left}, Key name A
OnKeyPress A
OnKeyup, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyup, Key code=65, Control keys=Shift{Left}, Key name A
OnKeyup, Key code=255, Control keys=Shift{Left}, Key name
OnKeyUp, Key code=16, Control keys=, Key name SHIFT
```

'3' results in:
```
OnKeyDown, Key code=255, Control keys=, Key name
OnKeyup, Key code=17, Control keys=, Key name CTRL
OnKeyup, Key code=17, Control keys=, Key name CTRL
OnKeyDown, Key code=51, Control keys=, Key name 3
OnkeyPress 3
OnKeyup, Key code=51, Control keys=, Key name 3
OnKeyup, Key code=17, Control keys=, Key name CTRL
OnKeyup, Key code=255, Control keys=, Key name
```

'#' results in:
```
OnKeyDown, Key code=16, Control keys=Shift{Left}, Key name SHIFT
OnKeyDown, Key code=255, Control keys=Shift{Left}, Key name
OnKeyup, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyup, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyDown, Key code=51, Control keys=Shift{Left}, Key name 3
OnKeyPress #
OnKeyup, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyup, Key code=51, Control keys=Shift{Left}, Key name A
OnKeyup, Key code=255, Control keys=Shift{Left}, Key name
OnKeyUp, Key code=16, Control keys=, Key name SHIFT
```

With a VM-inside-a-VM, 'a' gives:
```
OnKeyDown, Key code=255, Control keys=Shift{Left}, Key name
OnKeyUp, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyDown, Key code=255, Control keys=Shift{Left}, Key name
OnKeyUp, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyup, Key code=255, Control keys=Shift{Left}, Key name
OnKeyup, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyDown, Key code=64, Control keys=Shift{Left}, Key name A
OnKeyPress a
OnKeyup, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyup, Key code=65, Control keys=Shift{Left}, Key name A
```

'3' gives:
```
OnKeyDown, Key code=255, Control keys=Shift{Left}, Key name
OnKeyUp, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyDown, Key code=255, Control keys=Shift{Left}, Key name
OnKeyUp, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyup, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyup, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyDown, Key code=51, Control keys=Shift{Left}, Key name 3
OnKeyPress 3
OnKeyup, Key code=17, Control keys=Shift{Left}, Key name CTRL
OnKeyup, Key code=65, Control keys=Shift{Left}, Key name 3
```

**Comment by jbarber** [comment]
So in the nested VM situation, Shift appears permanently pressed.

**Comment by macocha** [comment]
Closed in #26
